snakemake_wrapper_utils/snakemake.py

```diff
@@ -11,14 +11,21 @@ def get_mem(snakemake, out_unit="MiB"):
     else:
         mem_mb = snakemake.resources.get("mem_mb", 205)
 
-    if out_unit == "KiB":
+    # Apply memory overhead
+    assert 0 <= mem_overhead_factor < 1, f"mem_overhead_factor must be between 0 and 1, got {mem_overhead_factor}"
+    mem_mb *= 1 - mem_overhead_factor
```

Comment:
I think you might want to do rounding here, because you'll otherwise get floating point numbers instead of the expected integer?

Here's the respective line for the java memory handling, where we first introduced this pattern:
https://github.com/snakemake/snakemake-wrapper-utils/blob/2cce1247feee4a1468f138abfde6f86c04f3a2ff/snakemake_wrapper_utils/java.py#L22

But thinking about this more, maybe we should actually make this [`floor()`](https://docs.python.org/3/library/math.html#math.floor) instead of `round()`, because rounding up might actually make the memory overhead reservation smaller than specified by the factor.